src/Stratis.Features.FederatedPeg/FederationGatewayFeature.cs

```diff
@@ -126,6 +130,15 @@ internal class FederationGatewayFeature : FullNodeFeature
 
         public override async Task InitializeAsync()
         {
+            if (this.federationManager?.CurrentFederationKey?.PubKey != null &&
+                this.federationManager.CurrentFederationKey.PubKey.ToString() != this.federationGatewaySettings.PublicKey)
+            {
+                this.logger.LogError("The '-publickey={0}' mismatches `{1}` as derived from the `{2}` file.",
```

Comment:
Shouldn't we throw afterwards? 

Also why taking 4 lines instead of 2?

Reply:
> Shouldn't we throw afterwards?

@quantumagi can you please answer?

Reply:
There are some cases where we don't throw after logging an error. If throwing will achieve something useful I am all for it. At this point we are just displaying a message on the console that stands out as an error.